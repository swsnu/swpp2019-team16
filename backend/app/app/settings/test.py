from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
