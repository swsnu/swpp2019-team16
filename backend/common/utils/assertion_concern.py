class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
