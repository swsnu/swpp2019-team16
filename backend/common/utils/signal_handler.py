import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
