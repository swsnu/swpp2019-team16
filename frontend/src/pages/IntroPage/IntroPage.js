import React from 'react';
import Main from '../_layouts/Main';

function IntroPage() {
  return <Main>Intro Page!</Main>;
}
export default IntroPage;
