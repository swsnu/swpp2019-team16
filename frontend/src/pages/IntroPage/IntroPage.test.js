import React from 'react';
import IntroPage from './IntroPage';
import { MemoryRouter, Route } from 'react-router-dom';
import { renderWithRedux } from '../../test/utils';

jest.mock('@fullpage/react-fullpage', () => ({ children }) => (
  <div>{children}</div>
));

describe('<IntroPage />', () => {
  const state = {
    auth: {
      auth: {
        id: 1,
      },
    },
    user: {
      user: null,
    },
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/intro']}>
        <Route component={IntroPage} path="/intro" />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
