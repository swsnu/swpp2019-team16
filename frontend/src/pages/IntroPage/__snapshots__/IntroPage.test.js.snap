// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroPage /> SHOULD match with snapshot 1`] = `
<div>
  <div>
    Intro Page!
  </div>
</div>
`;
